### PR TITLE
Solars panels can now be upgraded with glass

### PR DESCRIPTION
**Pull request by redmoogle**
# Document the changes in your pull request

image

also a tiny touchup on the glassless solar sprite

image

# Wiki Documentation

1500W Unupgraded
4500W Upgraded

Multipliers

Glass
1x Integ and Power

R. Glass
2x Integ 1.25x Power

Plasma
1.5x Integ 1.75x Power

R. Plasma
2.5x Integ 2x Power

Titanium
4x Integ 2.5x Power

Plastanium
5x Integ 3x Power

# Changelog

:cl:  
rscadd: Solar panels can be upgraded with glass
imageadd: Solar panels got a touch up
/:cl:

### CODE REVIEW

**Comment by Marmio64** [comment]
seems like denser glasses filled with metallic compounds would just block the sunlight and result in less energy, but when has ss13 ever made sense?

**Comment by redmoogle** [comment]
> seems like denser glasses filled with metallic compounds would just block the sunlight and result in less energy, but when has ss13 ever made sense?

It's just converting bluespace energies into usable photons :^)

File: code/modules/power/solar.dm [hunk 1]
```diff
@@ -21,21 +20,20 @@
 	var/obj/machinery/power/solar_control/control
 	var/needs_to_turn = TRUE //do we need to turn next tick?
 	var/needs_to_update_solar_exposure = TRUE //do we need to call update_solar_exposure() next tick?
-	var/obj/effect/overlay/panel
+	var/mutable_appearance/panelstructure
+	var/mutable_appearance/paneloverlay
+	var/multiplier
+	var/panelcolor
```
Comment (ToasterBiome) [inline code/modules/power/solar.dm:28]:
```suggestion
	var/obj/item/stack/sheet/glass_type

```

File: code/modules/power/solar.dm [hunk 2]
```diff
@@ -62,9 +60,13 @@
...
-	if(S.glass_type == /obj/item/stack/sheet/rglass) //if the panel is in reinforced glass
-		max_integrity *= 2 								 //this need to be placed here, because panels already on the map don't have an assembly linked to
-		obj_integrity = max_integrity
+	
+	S.glass_rating()
+
+	multiplier = S.multiplier
+	max_integrity *= S.integmultiplier
+	obj_integrity = max_integrity
+	panelcolor = S.panelcolor
```
Comment (ToasterBiome) [inline code/modules/power/solar.dm:76]:
```suggestion
	glass_type = S.glass_type
	
```

File: code/modules/power/solar.dm [hunk 3]
```diff
@@ -21,21 +20,20 @@
...
-	panel.icon = icon
-	panel.icon_state = "solar_panel"
-	panel.layer = FLY_LAYER
 	Make(S)
 	connect_to_network()
 	RegisterSignal(SSsun, COMSIG_SUN_MOVED, .proc/queue_update_solar_exposure)
+	panelstructure = mutable_appearance(icon, "solar_panel", FLY_LAYER)
+	paneloverlay = mutable_appearance(icon, "solar_panel-o", FLY_LAYER)
+	paneloverlay.color = panelcolor
+	update_icon()
```
Comment (ToasterBiome) [inline code/modules/power/solar.dm:38]:
```suggestion
/obj/machinery/power/solar/examine(mob/user)
	. = ..()
	if(glass_type)
		. += "It is covered in a sheet of [glass_type.name]"

```

**Review by ToasterBiome — COMMENTED** [review]
as Cartlord suggested this is an easy way to add what type of glass it has.

https://discord.com/channels/134720091576205312/205784753352343552/923888483142078474

Be aware that these are webedited reviews and thus cannot be used against me in a court of law (tm)

File: code/modules/power/solar.dm (hunk 1 above)
Comment (redmoogle) [reply code/modules/power/solar.dm:28]: this doesnt work.

Comment (ToasterBiome) [reply code/modules/power/solar.dm:28]: > this doesnt work.

oh really

Comment (redmoogle) [reply code/modules/power/solar.dm:28]: > > this doesnt work.
> 
> oh really

byond quirk where initial makes it valid for not a very logical reason

**Review by nesisknesis — APPROVED** [review]
hmmmm ok

**Review by JamieD1 — APPROVED** [review]
This is scientifically stupid

**Pull request merged**